mantidproject/mantid
# Use JobRunners for loading data in ALFView

**Description of work.**
This PR refactors some of the ALFView code to use a QtJobRunner to run algorithms. In aid of this, there is now an `ALFAlgorithmManager` class which handles the execution of each algorithm and calls back to the presenter.

This means that algorithm execution on ALFView is done on a background thread instead of a GUI thread. In a future PR, the GUI can be disabled before executing the algorithms.

A release note will be added in a future PR when the full conversion to background threading has been finished.

**To test:**
Open ALFView from the interface menu
Load ALF84116 as the sample
Load ALF78634 as the vanadium
Go to pick tab and select the `Select whole tube` button
Select the central four tubes (seen in the screenshot below)
The two theta value should be 38.066
Then click `Fit` on the right hand side of the interface
The Rotation angle should have a value of -0.1869...

![image](https://user-images.githubusercontent.com/40830825/216953757-def2ab6d-a0c2-4223-951b-f79753fe1e91.png)


It might be difficult to notice, but most of the processing is now done in a background process (excluding actions taken on the RHS of the GUI)

Part of #35114

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
